### PR TITLE
Getting Started and Other Docs

### PR DESCRIPTION
**Pull request by geoffcline**
**Issue, if available:** N/A

**Description of changes:**

paths to review:

website/content/en/docs/getting-started/_index.md 

do not review paths:

docs/* 

I respectfully submit that housing the HTML here makes PRs very confusing. Additionally, this code only works on local previews, not GH pages. 

**Screenshots**
karpenter docs

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/404.html [hunk 1]
```diff
@@ -3,36 +3,36 @@
   <head>
     <meta charset="utf-8">
 <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
-<meta name="generator" content="Hugo 0.85.0" /><META NAME="ROBOTS" CONTENT="NOINDEX, NOFOLLOW">
+<meta name="generator" content="Hugo 0.84.1" /><META NAME="ROBOTS" CONTENT="NOINDEX, NOFOLLOW">
```
Comment (njtran) [inline docs/404.html:6]: Is this referring to a version of Hugo? Is there a reason we wouldn't want to use the most recent version (seems like `0.86.0`

**Comment by akestner** [comment]
To get ahead of potential rabbit holes, I'm working on the hosting side of things now and will surface a PR with a recommendation

File: docs/docs/getting-started/eks-config.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+              - "ec2:TerminateInstances"
+              # Read Operations
+              - "ec2:DescribeLaunchTemplates"
+              - "ec2:DescribeInstances"
+              - "ec2:DescribeSecurityGroups"
+              - "ec2:DescribeSubnets"
+              - "ec2:DescribeInstanceTypes"
+              - "ec2:DescribeInstanceTypeOfferings"
+              - "ec2:DescribeAvailabilityZones"
+              - "ssm:GetParameter"
```
Comment (njtran) [inline docs/docs/getting-started/eks-config.yaml:40]: nit: can you add a new line?

File: website/content/en/docs/dev-guide.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+### Developer Loop
+* Make sure dependencies are installed
+    * Run `make codegen` to make sure yaml manifests are generated
+    * Run `make toolchain` to install cli tools for building and testing the project
+* You will need a personal development image repository (e.g. ECR)
+    * Make sure you have valid credentials to your development repository.
+    * `$KO_DOCKER_REPO` must point to your development repository
+    * Your cluster must have permissions to read from the repository
+* If you created your cluster on version 1.19 or above, you may need to tag your subnets as mentioned [here](docs/aws/README.md). This is a temporary problem with our subnet discovery system, and is being tracked [here](https://github.com/awslabs/karpenter/issues/404#issuecomment-845283904).
```
Comment (njtran) [inline website/content/en/docs/dev-guide.md:38]: We can remove this line as we have instructions in the getting_started doc to handle this.

File: website/content/en/docs/faq.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+---
+title: "FAQs"
+linkTitle: "FAQs"
+weight: 30
+---
+
+## General
+### How does a Provisioner decide to manage a particular node?
+Each node will have a set of predetermined Karpenter labels. Provisioners will use `name` and `namespace` labels to distinguish between Provisioners. Furthermore, a Provisioner will only take action on a node based on the label that details what phase a node is in. e.g. a provisioner will only consider a node for termination if its phase label says "underutilized".
```
Comment (njtran) [inline website/content/en/docs/faq.md:9]: Would you be able to change this to be the following?
> Provisioners will use a `name` label to distinguish between Provisioners

File: website/content/en/docs/getting-started/_index.md [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+  https://api.github.com/repos/awslabs/karpenter/releases/latest \
+  | jq -r '.tag_name')
+```
+
+### Create a Cluster
+
+Create a cluster with `eksctl`. The [example configuration](eks-config.yaml) file specifies a basic cluster (name, region), and an IAM role for Karpenter to use. 
+
+```bash
+## This command won't work until a release after v0.2.8 is released
```
Comment (njtran) [inline website/content/en/docs/getting-started/_index.md:60]: Feel free to remove this comment

File: website/content/en/docs/getting-started/_index.md [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+curl -fsSL  https://raw.githubusercontent.com/awslabs/karpenter/"${KARPENTER_VERSION}"/pkg/cloudprovider/aws/docs/eks-config.yaml \
```
Comment (njtran) [inline website/content/en/docs/getting-started/_index.md:61]: Will this still be hosted in Github? Or in the website?

File: website/content/en/docs/getting-started/_index.md [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+```
+
+Now, Karpenter can send requests for new EC2 instances to AWS. 
+
+### Install Karpenter Helm Chart
+
+Use helm to deploy Karpenter to the cluster. 
+
+We created a Kubernetes service account when we created the cluster using
+eksctl. Thus, we don't need the helm chart to do that.
```
Comment (njtran) [inline website/content/en/docs/getting-started/_index.md:116]: Great. WDYT about having a command here (or a link to a command) to patch the serviceAccount for BYOC experiences or debugging purposes?

File: website/content/en/docs/getting-started/_index.md [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+shapes. In other words, Karpenter eliminates the need to manage many different
+node groups. Karpenter makes scheduling and provisioning decisions based on pod
+attributes such as labels and affinity. 
+
+Create a simple default provisioner using the command below. This provisioner
+provides instances with the default certificate bundle, and the control plane
+endpoint url. 
+
+Importantly, the `ttlSecondsAfterEmpty` value configures Karpenter to
+deprovision empty nodes.
```
Comment (njtran) [inline website/content/en/docs/getting-started/_index.md:137]: It may be worth to include `TTLSecondsUntilExpired` here as well (which recycles nodes automatically after some TTL). 

Additionally we could include the fact if someone doesn't want these two features, they can simply leave it out of the provisioner spec.

File: website/content/en/docs/getting-started/_index.md [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+```
+
+## First Use
+
+Karpenter is now active and ready to begin provisioning nodes. Create a
+workload (e.g., deployment) to see Karpenter provision some nodes. 
+
+### Create Workload
+
+This deployment uses the [pause image](https://www.ianlewis.org/en/almighty-pause-container) and starts with zero replicas.
```
Comment (njtran) [inline website/content/en/docs/getting-started/_index.md:162]: We also have the feature to include a `karpenter.sh/do-not-evict:true` annotation for pods they don't want to evict. This is a pretty specific feature, so it may overcomplicate this guide. WDYT?

File: website/content/en/docs/getting-started/_index.md [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+Karpenter should deprovision the now empty nodes. 
+
+```bash
+kubectl delete deployment inflate
+kubectl logs -f -n karpenter $(kubectl get pods -n karpenter -l karpenter=controller -o name)
+```
+
+### Manual Node Deprovisioning
+
+If you delete a node with kubectl, Karpenter deprovisions the corresponding instance.
```
Comment (njtran) [inline website/content/en/docs/getting-started/_index.md:210]: nit: deprovisions -> terminates

File: website/content/en/docs/getting-started/eks-config.yaml [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+              - "ec2:TerminateInstances"
+              # Read Operations
+              - "ec2:DescribeLaunchTemplates"
+              - "ec2:DescribeInstances"
+              - "ec2:DescribeSecurityGroups"
+              - "ec2:DescribeSubnets"
+              - "ec2:DescribeInstanceTypes"
+              - "ec2:DescribeInstanceTypeOfferings"
+              - "ec2:DescribeAvailabilityZones"
+              - "ssm:GetParameter"
```
Comment (njtran) [inline website/content/en/docs/getting-started/eks-config.yaml:40]: nit: new line

**Review by njtran — COMMENTED** [review]
Nice work! Just a couple of nits and possible suggestions.

File: website/content/en/docs/faq.md (hunk 4 above)
Comment (akestner) [inline website/content/en/docs/faq.md:9]: Can we be more specific about the labels that Karpenter will add to nodes?

Comment (akestner) [inline website/content/en/docs/faq.md:9]:
```suggestion
Each node will have a set of predetermined Karpenter labels. Provisioners will use `name` and `namespace` labels to distinguish between Provisioners. Furthermore, a Provisioner will only take action on a node based on the label that details what phase a node is in. e.g. a provisioner will only consider a node for termination if its phase label says "underutilized".
```
```suggestion
Each node will have a set of predetermined Karpenter labels. Provisioners will use `name` and `namespace` labels to distinguish between Provisioners. Furthermore, a Provisioner will only take action on a node based on the label that details what phase a node is in. e.g. a provisioner will only consider a node for termination if its phase label says "underutilized".
```
```suggestion
Each node will have a set of predetermined Karpenter labels. Provisioners will use the `name` and `namespace` labels to distinguish between Provisioners. Furthermore, a Provisioner will only take action on a node based on the label that details what phase a node is in, e.g. a Provisioner will only consider a node for termination if its phase label says `"underutilized"`.
```

File: website/content/en/docs/faq.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+### Does Karpenter replace the Kube Scheduler?
+No. Provisioners work in tandem with the Kube Scheduler. When capacity is unconstrained, the Kube Scheduler will schedule pods as usual. It may schedule pods to nodes managed by Provisioners or other types of capacity in the cluster. Provisioners only attempt to schedule pods when `type=PodScheduled,reason=Unschedulable`. In this case, they will make a provisioning decision, launch new capacity, and bind pods to the provisioned nodes. Provisioners do not wait for the Kube Scheduler to make a scheduling decision in this case, as the decision is already made by nature of making a provisioning decision. It's possible that a node from another management solution could create a race between the kube scheduler and Karpenter. In this case, the first binding call will win. If Karpenter loses this race, the node will eventually be cleaned up.
```
Comment (akestner) [inline website/content/en/docs/faq.md:14]:
```suggestion
No. Provisioners work in tandem with the Kube Scheduler. When capacity is unconstrained, the Kube Scheduler will schedule pods as usual. It may schedule pods to nodes managed by Provisioners or other types of capacity in the cluster. Provisioners only attempt to schedule pods when `type=PodScheduled,reason=Unschedulable`. In this case, they will make a provisioning decision, launch new capacity, and bind pods to the provisioned nodes. Provisioners do not wait for the Kube Scheduler to make a scheduling decision in this case, as the decision is already made by nature of making a provisioning decision. It's possible that a node from another management solution, like the Cluster Autoscaler, could create a race between the `kube-scheduler` and Karpenter. In this case, the first binding call will win, although Karpenter will often win these race conditions due to its performance characteristics. If Karpenter loses this race, the node will eventually be cleaned up.
```

File: website/content/en/docs/faq.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+### Does Karpenter support node selectors?
+Yes. Node selectors are an opt-in mechanism which allows customers to specify the nodes to which a pod can schedule. Provisioners recognize well known node selectors on incoming pods and use them to constrain the nodes they generate. For example, well known selectors like `node.kubernetes.io/instance-type`, `topology.kubernetes.io/zone`, `kubernetes.io/os`, `kubernetes.io/arch` are supported, and will ensure that provisioned nodes are constrained accordingly. Additionally, customers may specify arbitrary labels, which will be automatically applied to every node launched by the Provisioner.
```
Comment (akestner) [inline website/content/en/docs/faq.md:16]:
```suggestion
Yes. Node selectors are an opt-in mechanism which allow customers to specify the nodes on which a pod can scheduled. Provisioners recognize well-known node selectors on incoming pods and use them to constrain the nodes they generate. You can read more about the well-known node selectors Karpenter supports in the [Concepts](/docs/concepts/#well-known-labels) documentation. For example, well known selectors like `node.kubernetes.io/instance-type`, `topology.kubernetes.io/zone`, `kubernetes.io/os`, `kubernetes.io/arch` are supported, and will ensure that provisioned nodes are constrained accordingly. Additionally, customers may specify arbitrary labels, which will be automatically applied to every node launched by the Provisioner.
```

File: website/content/en/docs/faq.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+### Does Karpenter support taints?
+Yes. Taints are an opt-out mechanism which allows customers to specify the nodes to which a pod cannot schedule. Unlike labels, Provisioners do not automatically taint nodes in response to pod tolerations, since pod tolerations do not require that corresponding taints exist. However, similar to labels, customers may specify taints for their Provisioner, which will automatically be applied to every node in the group. This means that if a Provisioner is configured with taints, any incoming pods will not be provisioned unless they tolerate the taints.
```
Comment (akestner) [inline website/content/en/docs/faq.md:18]:
```suggestion
Yes. Taints are an opt-out mechanism which allows customers to specify the nodes on which a pod cannot be scheduled. Unlike labels, Provisioners do not automatically taint nodes in response to pod tolerations, since pod tolerations do not require that corresponding taints exist. However, similar to labels, customers may specify taints for their Provisioner, which will automatically be applied to every node in the group. This means that if a Provisioner is configured with taints, any incoming pods will not be provisioned unless they tolerate the taints.
```

File: website/content/en/docs/faq.md [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+### Does Karpenter support topology spread constraints?
+Yes. Provisioners respect `pod.spec.topologySpreadConstraints`. Allocating pods with these constraints may yield highly fragmented nodes, due to their strict nature and complexity of “online binpacking” algorithms. However, the reallocation pass is able to produce much more efficient packings using “offline binpacking” techniques.
```
Comment (akestner) [inline website/content/en/docs/faq.md:20]:
```suggestion
Yes. Provisioners respect `pod.spec.topologySpreadConstraints`. Allocating pods with these constraints may yield highly fragmented nodes, due to their strict nature and complexity of “online binpacking” algorithms.
```

File: website/content/en/docs/faq.md [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+### Does Karpenter support affinity?
+No. Karpenter intentionally does not support affinity due the to [scalability limitations](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity) outlined by SIG Scalability. Instead, we recommend using node selectors or taints instead of node affinity and pod topology spread instead of pod affinity. Do you have a use case for affinity that we're missing? We're excited to hear about it in our [Working Group](working-group/README.md).
```
Comment (akestner) [inline website/content/en/docs/faq.md:22]:
```suggestion
No. Karpenter intentionally does not support affinity due the to [scalability limitations](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity) outlined by SIG Scalability. Instead, we recommend using node selectors or taints instead of node affinity and pod topology spread instead of pod affinity. Do you have a use case for affinity that we're missing? Open an issue in our [GitHub repo](https://github.com/awslabs/karpenter/issues/new/choose) and tell us about it!
```

File: website/content/en/docs/faq.md [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+### Does Karpenter support custom resource like accelerators or HPC?
+Yes. Support for specific custom resources can be implemented by your cloud provider.
+### Does Karpenter support daemonsets?
+Yes. Provisioners factor in daemonset overhead into all allocation and reallocation calculations. They also respect daemonset scheduling constraints, such as Nvidia’s GPU Driver Installer.
```
Comment (akestner) [inline website/content/en/docs/faq.md:26]:
```suggestion
Yes. Provisioners factor in daemonset overhead into all allocation calculations. They also respect daemonset scheduling constraints, such as Nvidia’s GPU Driver Installer.
```

File: website/content/en/docs/faq.md [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+### Does Karpenter support multiple scheduling defaults?
+Provisioners are heterogeneous, which means that the nodes they manage are spread across multiple availability zones, instance types, and capacity types. This flexibility reduces the need for a large number of groups. However, customers may find multiple groups to be useful for more advanced use cases. For example, customers can create multiple groups, and then use the node selector `karpenter.sh/provisioner-name` to target specific groups. This enables advanced use cases like resource isolation and sharding.
+### What if my pod is schedulable for multiple Provisioners?
+It's possible that an unconstrained pods could flexibly schedule in multiple groups. In this case, Provisioners will race to create a scheduling lease for the pod before launching new nodes, which avoids unnecessary scale out.
```
Comment (akestner) [inline website/content/en/docs/faq.md:30]:
```suggestion
It's possible that unconstrained pods could flexibly schedule in multiple groups. In this case, Provisioners will race to create a scheduling lease for the pod before launching new nodes, which avoids unnecessary scale out.
```

File: website/content/en/docs/faq.md [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+## Reallocation
```
Comment (akestner) [inline website/content/en/docs/faq.md:31]: This section feels mis-titled, I was expecting to hear about how Karpenter improves cluster efficiency over time.
```suggestion
## Deprovisioning
```

File: website/content/en/docs/faq.md [hunk 20]
```diff
@@ -0,0 +1,48 @@
+---
+title: "FAQs"
+linkTitle: "FAQs"
+weight: 30
+---
+
+## General
+### How does a Provisioner decide to manage a particular node?
+Each node will have a set of predetermined Karpenter labels. Provisioners will use `name` and `namespace` labels to distinguish between Provisioners. Furthermore, a Provisioner will only take action on a node based on the label that details what phase a node is in. e.g. a provisioner will only consider a node for termination if its phase label says "underutilized".
+## Allocation
```
Comment (akestner) [inline website/content/en/docs/faq.md:10]:
```suggestion
## Provisioning
```

File: website/content/en/docs/faq.md [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+Karpenter annotates nodes that are underutilized with a time to live (TTL). If the node remains underutilized after the TTL expires, Karpenter then [cordons](https://kubernetes.io/docs/concepts/architecture/nodes/#manual-node-administration) the node and uses the [Kubernetes Eviction API](https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/#eviction-api) to evict all non-daemonset pods. Once the node is empty, the node is terminated.
+### Does Karpenter support Pod Disruption Budgets?
+Yes. The Kubernetes Eviction API will not delete pods that violate a [Pod Disruption Budget (PDB)](https://kubernetes.io/docs/tasks/run-application/configure-pdb/). It also disallows eviction of any pod covered by multiple PDBs, so most users will want to avoid overlapping selectors. See [this](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets) for more.
+### Does Karpenter support scale to zero?
+Yes. Provisioners start at zero and launch or terminate nodes as necessary. We recommend that customers maintain a small amount of static capacity to bootstrap system controllers or run Karpenter outside of their cluster.
+## Compatibility
```
Comment (akestner) [inline website/content/en/docs/faq.md:42]: Let's bump this section up since these are more common questions than some of the more prominently placed ones.

File: website/content/en/docs/faq.md [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+### Can I use Karpenter alongside another node management solution?
+Provisioners are designed to work alongside static capacity management solutions like EKS Managed Node Groups and EC2 Auto Scaling Groups. Some customers may choose to (1) manage the entirety of their capacity using Provisioner, others may prefer (2) a mixed model with both dynamic and statically managed capacity, some may prefer (3) a fully static approach. We anticipate that most customers will fall into bucket (2) in the short term, and (1) in the long term.
+### Can I use Karpenter with the Kubernetes Cluster Autoscaler?
+Yes, with side effects. Karpenter is a Cluster Autoscaler replacement. Both systems scale up nodes in response to pending pods. If configured together, both systems will race to launch new instances for incoming pods. Since Provisioners make binding decisions, they will typically win the scheduling race. In this case, the Cluster Autoscaler will eventually scale down the unnecessary capacity. If the Cluster Autoscaler is configured with Node Groups that have constraints that aren’t supported by any Provisioner, its behavior will continue unimpeded.
```
Comment (akestner) [inline website/content/en/docs/faq.md:48]:
```suggestion
Yes, with side effects. Karpenter is a Cluster Autoscaler replacement. Both systems scale up nodes in response to unschedulable pods. If configured together, both systems will race to launch new instances for these pods. Since Provisioners make binding decisions, Karpenter will typically win the scheduling race. In this case, the Cluster Autoscaler will eventually scale down the unnecessary capacity. If the Cluster Autoscaler is configured with Node Groups that have constraints that aren’t supported by any Provisioner, its behavior will continue unimpeded.
```

File: website/content/en/search.md [hunk 23]
```diff
@@ -0,0 +1,6 @@
+---
+title: Search Results
```
Comment (akestner) [inline website/content/en/search.md:2]: Is this file necessary? How are we handling search in this PR? 

While I think it'll be useful to include a way for users to search the docs for the content they're looking for, I'm inclined to defer that work to a separate PR as this is PR is getting quite lengthy to begin with.

File: website/package-lock.json [hunk 24]
```diff
@@ -1,8 +1,931 @@
 {
   "name": "website",
   "version": "1.0.0",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
   "requires": true,
+  "packages": {
```
Comment (akestner) [inline website/package-lock.json:6]: nit: What cause the changes to this file?

File: website/public/404.html [hunk 25]
```diff
@@ -0,0 +1,128 @@
+<!doctype html>
```
Comment (akestner) [inline website/public/404.html:1]: I think we should consider removing the build generated files under `public/` as well as those built under `../docs/` since they cause a ton of noise in these PRs. Thoughts?

Even if we decide to keep these built files in the PR, there's no reason we'd need to have both these in `public/` and those in `../docs/`

**Review by akestner — CHANGES_REQUESTED** [review]
Great work! I think this is definitely on the right track and a huge improvement over the current in-repo docs. Let's keep pushing on these to get them completed and hopefully we can get this published before the Bug Bash tomorrow.

File: docs/404.html (hunk 1 above)
Comment (geoffcline) [reply docs/404.html:6]: this is mooted by #528 build on netlify

File: website/content/en/docs/getting-started/_index.md (hunk 9 above)
Comment (geoffcline) [reply website/content/en/docs/getting-started/_index.md:162]: i'll include that in the annotations section

File: website/content/en/docs/faq.md (hunk 4 above)
Comment (geoffcline) [reply website/content/en/docs/faq.md:9]: ty

File: website/content/en/docs/getting-started/_index.md (hunk 6 above)
Comment (geoffcline) [reply website/content/en/docs/getting-started/_index.md:61]: sticking with this URL for now

File: website/content/en/docs/getting-started/_index.md (hunk 7 above)
Comment (geoffcline) [reply website/content/en/docs/getting-started/_index.md:116]: i agree but i think that's out of scope for the PR right now.

File: website/content/en/docs/getting-started/_index.md (hunk 8 above)
Comment (geoffcline) [reply website/content/en/docs/getting-started/_index.md:137]: added some desc of expiration and a link to the prov-crd file

File: website/content/en/docs/dev-guide.md (hunk 3 above)
Comment (geoffcline) [reply website/content/en/docs/dev-guide.md:38]: I'm not sure what exactly you are referring to here?

File: website/content/en/docs/_index.md [hunk 26]
```diff
@@ -0,0 +1,11 @@
...
+---
+title: "Documentation"
+linkTitle: "Documentation"
+weight: 20
+menu:
+  main:
+    weight: 20
+---
+
+
```
[the remark this answers is not in the document]
Comment (geoffcline) [reply website/content/en/docs/_index.md:11]: can this be deferred until the next PR?

File: website/content/en/docs/dev-guide.md [hunk 27]
```diff
@@ -0,0 +1,86 @@
+---
+title: "Development Guide"
```
[the remark this answers is not in the document]
Comment (geoffcline) [reply website/content/en/docs/dev-guide.md:2]: agreed. done.

`dev-guide.md` -> `development-guide.md`
`prov-crd.md` -> `provisioner-crd.md`

File: website/content/en/docs/faq.md [hunk 28]
```diff
@@ -0,0 +1,48 @@
+---
+title: "FAQs"
```
[the remark this answers is not in the document]
Comment (geoffcline) [reply website/content/en/docs/faq.md:2]: done

File: website/content/en/search.md (hunk 23 above)
Comment (geoffcline) [reply website/content/en/search.md:2]: this is merely the search template, i'll remove it.

I'm planning on using lunr.js, as it's simple and integrated with hugo.

File: website/content/en/docs/faq.md (hunk 4 above)
Comment (geoffcline) [reply website/content/en/docs/faq.md]: i'll prep an example tomorrow, unless @njtran has an example before then? :)

File: website/content/en/docs/faq.md (hunk 21 above)
Comment (geoffcline) [reply website/content/en/docs/faq.md]: moved `### Does Karpenter support scale to zero?` to under `## General`

File: website/package-lock.json (hunk 24 above)
Comment (geoffcline) [reply website/package-lock.json:6]: unsure? something in the makefile?

File: website/public/404.html (hunk 25 above)
Comment (geoffcline) [reply website/public/404.html]: i have removed `website/public`

we will remove `../docs/` in the netlify pr (or a follow up)

File: website/content/en/docs/dev-guide.md (hunk 3 above)
Comment (njtran) [reply website/content/en/docs/dev-guide.md]: In our getting started guide we give a command to tag subnets, so as long as readers follow that guide, they should be fine.

https://github.com/awslabs/karpenter/blob/main/pkg/cloudprovider/aws/docs/README.md#create-a-cluster

File: website/content/en/docs/faq.md (hunk 4 above)
Comment (njtran) [reply website/content/en/docs/faq.md]: The only labels that Karpenter will add into a node are listed [here](https://github.com/awslabs/karpenter/blob/main/pkg/apis/provisioning/v1alpha3/provisioner.go#L120-L126). The do-not-evict annotation will not be added by Karpenter but is respected in eviction.

Comment (njtran) [reply website/content/en/docs/faq.md]: Additionally, we will add the finalizer, listed in the same file [here](https://github.com/awslabs/karpenter/blob/main/pkg/apis/provisioning/v1alpha3/provisioner.go#L133).

Also, any constraints specified in the provisioner will be added to the nodes on allocation.

**Comment by akestner** [comment]
Hey @geoffcline, it looks like you've force pushed to this branch a few times since it was reviewed. Is there something that you're having trouble with?

Force pushing to branches that have been shared publicly is generally frowned upon because it rewrites the git repo's history, making it difficult to understand what changes have been made and when. Back in the day, I kept [this](http://justinhileman.info/article/git-pretty/full/) printed out next to my desk for when I got into sticky situations.

Let me know if I can help, we can touch base tomorrow as needed.

**Comment by geoffcline** [comment]
I was doing further development and forgot that those changes were showing up here. 

Kubernetes website instructs users to force push with each new commit on a PR (they want 1 commit for each PR), so it's a bit of a learning curve for a different project.

File: website/content/en/docs/faq.md (hunk 21 above)
Comment (akestner) [reply website/content/en/docs/faq.md]: I was referring to the `## Compatibility` section, not scale to zero

**Comment by akestner** [comment]
Ahh I see, glad to hear all's well. We're getting close here, giving this another look now

File: website/package-lock.json (hunk 24 above)
Comment (akestner) [reply website/package-lock.json:6]: Alright, nbd

File: website/content/en/docs/faqs.md [hunk 29]
```diff
@@ -0,0 +1,53 @@
...
+
+## General
+### How does a Provisioner decide to manage a particular node?
+<<<<<<< HEAD:website/content/en/docs/faq.md
+Each node will have a set of predetermined Karpenter labels. Provisioners will use the `name` and `namespace` labels to distinguish between Provisioners. Furthermore, a Provisioner will only take action on a node based on the label that details what phase a node is in, e.g. a Provisioner will only consider a node for termination if its phase label says `"underutilized"`.
+## Provisioning
+=======
+Each node will have a set of predetermined Karpenter labels. Provisioners will use a `name` label to distinguish between Provisioners. Furthermore, a Provisioner will only take action on a node based on the label that details what phase a node is in. e.g. a provisioner will only consider a node for termination if its phase label says "underutilized".
+## Allocation
+>>>>>>> eaa48cf (update filenames and get started guide):website/content/en/docs/faqs.md
```
Comment (geoffcline) [inline website/content/en/docs/faqs.md:15]: just saw this -- will fix asap

File: website/content/en/docs/faq.md (hunk 4 above)
Comment (akestner) [reply website/content/en/docs/faq.md]: Can't suggest here, I'm going to resolve and then add what i had in mind in the current file

File: website/content/en/docs/faq.md (hunk 21 above)
Comment (geoffcline) [reply website/content/en/docs/faq.md]: ah ty

Comment (geoffcline) [reply website/content/en/docs/faq.md]: moved compatibility to the section after general.

File: website/content/en/docs/_index.md (hunk 26 above)
[the remark this answers is not in the document]
Comment (akestner) [reply website/content/en/docs/_index.md:11]: How about something like this?

```suggestion
Karpenter is an open-source autoscaling project built for Kubernetes. It improves availability for Kubernetes applications without requiring manually or over-provisioning compute resources. Karpenter is designed to provide the right compute resources to match your application’s needs in seconds, instead of minutes by observing the aggregate resource requests of unschedulable pods and makes decisions to launch and terminate nodes to minimize scheduling latencies.

Learn more about Karpetner and how to get started below.
```

File: website/content/en/docs/provisioner-crd.md [hunk 30]
```diff
@@ -0,0 +1,42 @@
+---
+title: "Provisioner CRD"
+linkTitle: "Provisioner CRD"
+weight: 40
+date: 2017-01-05
+---
+
+## example resource
```
Comment (akestner) [inline website/content/en/docs/provisioner-crd.md:8]:
```suggestion
## Example Provisioner Resource
```

**Review by akestner — CHANGES_REQUESTED** [review]
I think we're really close, couple minor suggestions and I think we'll be good to go!

**Comment by akestner** [comment]
Ship it!

**Review by akestner — APPROVED** [review]
